src/lustre/lustreTransSys.ml

```diff
@@ -489,7 +489,7 @@ let add_constraints_of_type init terms state_var =
 
     (* Get bounds of integer range *)
     let l, u = 
-      (if Type.is_array state_var_type 
+      (if (Type.is_array state_var_type || Type.is_int_range state_var_type)
```

Comment:
At this location `Type.is_array state_var_type` is always false. The condition should be simply:
```
Type.is_int_range state_var_type
```

Reply:
Addressed